### PR TITLE
Add BeagleConnect Freedom

### PR DESCRIPTION
**Pull request by jadonk**
BeagleConnect Freedom is an open hardware development platform featuring 2x mikroBUS interfaces and utilizes a Texas Instruments (TI) CC1352P7 microcontroller which features an ARM Cortex-M4 CPU, 704kB flash, 2.4GHz radio, and SubGHz radio.

See https://beagleconnect.org

### CODE REVIEW

**Comment by vaishnavachath** [comment]
@jadonk , Awesome, Thank you for the Pull request, hal_ti needs to be updated to have base support for P7 series right ? and those changes needs to be merged prior to this PR.Is there a PR already open under hal_ti for this?

**Comment by jadonk** [comment]
> @jadonk , Awesome, Thank you for the Pull request, hal_ti needs to be updated to have base support for P7 series right ? and those changes needs to be merged prior to this PR.Is there a PR already open under hal_ti for this?

Yes, based on your version of the changes necessary.

**Comment by cfriedt** [comment]
Judging by this, it looks like the PR for `hal_ti` might need to be revised to a more recent commit. This issue was solved by not redeclaring `pthread_self()` inside of `hal_ti`.

```
2023-03-07T07:47:08.3944050Z ccache /opt/toolchains/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DKERNEL -DSL_PLATFORM_MULTI_THREADED -DSL_SUPPORT_IPV6 -DUSE_CC3220_ROM_DRV_API -D_ANSI_SOURCE -D_FORTIFY_SOURCE=1 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -Dgcc -I/__w/zephyr/zephyr/include -Izephyr/include/generated -I/__w/zephyr/zephyr/soc/arm/ti_simplelink/cc32xx -I/__w/zephyr/zephyr/lib/libc/newlib/include -I/__w/zephyr/zephyr/lib/posix/getopt/. -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I/__w/zephyr/modules/hal/ti/simplelink/source/ti/devices/cc32xx/. -I/__w/zephyr/modules/hal/ti/simplelink/source/ti/devices/cc32xx/inc -I/__w/zephyr/modules/hal/ti/simplelink/source/ti/devices/cc32xx/driverlib -I/__w/zephyr/modules/hal/ti/simplelink/. -I/__w/zephyr/modules/hal/ti/simplelink/source -I/__w/zephyr/modules/hal/ti/simplelink/kernel/zephyr/dpl -I/__w/zephyr/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/porting -fno-strict-aliasing -Werror -Og -imacros /__w/zephyr/zephyr/twister-out/cc3220sf_launchxl/samples/net/wifi/sample.net.wifi/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m4 -mthumb -mabi=aapcs --sysroot=/opt/toolchains/zephyr-sdk-0.15.2/arm-zephyr-eabi/arm-zephyr-eabi -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/samples/net/wifi=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -D_POSIX_THREADS -std=c99 -MD -MT modules/ti/simplelink/CMakeFiles/..__modules__hal__ti__simplelink.dir/source/ti/drivers/net/wifi/source/netapp.c.obj -MF modules/ti/simplelink/CMakeFiles/..__modules__hal__ti__simplelink.dir/source/ti/drivers/net/wifi/source/netapp.c.obj.d -o modules/ti/simplelink/CMakeFiles/..__modules__hal__ti__simplelink.dir/source/ti/drivers/net/wifi/source/netapp.c.obj -c /__w/zephyr/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/source/netapp.c
2023-03-07T07:47:08.3948847Z In file included from /__w/zephyr/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/source/netapp.c:44:
2023-03-07T07:47:08.3950025Z /__w/zephyr/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/source/driver.h:463:15: error: conflicting types for 'pthread_self'; have 'void *(void)'
2023-03-07T07:47:08.3950609Z   463 | extern void * pthread_self(void);
2023-03-07T07:47:08.3950996Z       |               ^~~~~~~~~~~~
2023-03-07T07:47:08.3951504Z In file included from /__w/zephyr/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/cc_pal.h:54,
2023-03-07T07:47:08.3952131Z                  from /__w/zephyr/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h:50,
2023-03-07T07:47:08.3952739Z                  from /__w/zephyr/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/simplelink.h:243,
2023-03-07T07:47:08.3953509Z                  from /__w/zephyr/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/source/netapp.c:42:
2023-03-07T07:47:08.3954465Z /__w/zephyr/zephyr/include/zephyr/posix/pthread.h:419:11: note: previous declaration of 'pthread_self' with type 'pthread_t(void)' {aka 'unsigned int(void)'}
2023-03-07T07:47:08.3955070Z   419 | pthread_t pthread_self(void);

```

File: west.yml
Comment (cfriedt) [inline west.yml:142]: revision should be `pull/45/head` here until the `hal_ti` PR is merged, but that cannot be merged until this PR is all green. After the `hal_ti` PR is merged, this can be changed to the new HEAD commit in `hal_ti`

**Review by cfriedt — CHANGES_REQUESTED** [review]
Looks OK to me mostly. Just minor remark about the `hal_ti` revision. Also, I'm not entirely sure if `.webp` is ok for image formats. @kartben - do you know? Most of our docs use images that are `.jpg` (or in some cases `.png`) and the size is usually limited to some 10's of kB.

**Comment by kartben** [comment]
> Looks OK to me mostly. Just minor remark about the `hal_ti` revision. Also, I'm not entirely sure if `.webp` is ok for image formats. @kartben - do you know? Most of our docs use images that are `.jpg` (or in some cases `.png`) and the size is usually limited to some 10's of kB.

I'm happy with having .webp images, especially since in this case that means having a really nice high-ish resolution, transparent, image. However, you may want to set the :width: of the figures to 500px as right now they are pretty big :)

File: west.yml
Comment (cfriedt) [reply west.yml:142]: @jadonk - kind of a weird little dance with module dependencies. Sorry for the delay. Documented here:

https://docs.zephyrproject.org/latest/develop/modules.html#submitting-changes-to-modules

Comment (jadonk) [reply west.yml]: I'm not sure how to mark it DNM, but I made the west.yml change to `pull/45/head`.

**Comment by jadonk** [comment]
I'm confused about the typedef issue. `pinctrl_soc_pin_t` is defined many times. It probably shouldn't be, but it is, so how is this supposed to be handled?

File: west.yml
Comment (jadonk) [reply west.yml]: I think I've handled these requests. For the typedefs, I hope you can make a suggestion as this must be a common problem.

**Review by vaishnavachath — CHANGES_REQUESTED** [review]
@jadonk can you fix the empty commit message body here on https://github.com/zephyrproject-rtos/zephyr/pull/55488/commits/aee655f778e62b81ea6fa8d1052ad965982a683a ?
https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-body

File: modules/Kconfig.simplelink
Comment (vaishnavachath) [inline modules/Kconfig.simplelink:35]: @jadonk can you please drop this newline at EOF.

Comment (jadonk) [reply modules/Kconfig.simplelink:35]: k.

**Comment by vaishnavachath** [comment]
> I'm confused about the typedef issue. `pinctrl_soc_pin_t` is defined many times. It probably shouldn't be, but it is, so how is this supposed to be handled?

@jadonk I think we need to fix it in the following manner?
```
-typedef struct pinctrl_soc_pin {
+struct pinctrl_soc_pin {
        uint32_t pin;
        uint32_t iofunc;
        uint32_t iomode;
-} pinctrl_soc_pin_t;
+};
+
+typedef struct pinctrl_soc_pin pinctrl_soc_pin_t;
```

File: west.yml
Comment (cfriedt) [inline west.yml:142]: the `hal_ti` PR has been merged. You can go ahead and updated this to 4e0e053f5ea413ef0cb20000729038d921ac2746

**Review by cfriedt — CHANGES_REQUESTED** [review]
Should be ready to merge after
1. `west.yml` is changed (set revision to 4e0e053f5ea413ef0cb20000729038d921ac2746). 
2. typedef / struct issue is resolved

Hopefully, Vaishnav's suggestion will get around the compliance issue. Otherwise, we will need someone with superpowers to merge.

File: west.yml
Comment (jadonk) [reply west.yml:142]: I'll pull it in with a rebase.

**Comment by jadonk** [comment]
> I'm going to take a look at devices with pinctrl_soc.h, but without a typedef:
> 
> ```
> jkridner@slotcar:~/workspace/zephyr$ fgrep -L typedef `find . | grep pinctrl_soc.h`
> ./soc/arm/nxp_imx/rt/pinctrl_soc.h
> ./soc/arm/nxp_s32/s32ze/pinctrl_soc.h
> ./soc/arm/atmel_sam0/common/pinctrl_soc.h
> ./soc/arm/atmel_sam/common/pinctrl_soc.h
> ./soc/arm/gigadevice/common/pinctrl_soc.h
> ./soc/riscv/riscv-privilege/gd32vf103/pinctrl_soc.h
> ```

nxp_imx: nope, just includes one of 2 .h files that do the same thing.
nxp_s32: nope, just includes another .h file to do the same thing.
atmel_sam0: nope, just includes another .h file to do the same thing.
atmel_sam: nope
gigadevice: nope
riscv-priviledge: nope

So, how is anyone adding new SoCs with pinctrl support?

File: soc/arm/ti_simplelink/cc13x2x7_cc26x2x7/soc.h
Comment (cfriedt) [inline soc/arm/ti_simplelink/cc13x2x7_cc26x2x7/soc.h:10]: Zephyr has a CMSIS module - could this be pulled in from elsewhere, avoiding the additional typedef?

That's the last compliance issue.

**Comment by jadonk** [comment]
Not sure why it is saying I cancelled review requests. I welcome all reviews. :-D

**Review by cfriedt — APPROVED** [review]
Beautiful! Excellent work @jadonk, @vaishnavachath, et al at BeagleBoard

How do I buy one of the newest rev boards?

**Comment by jadonk** [comment]
https://beagleconnect.org :-D

**Review by vaishnavachath — APPROVED** [review]

**Pull request merged**
